File: src/main/java/loganalazyer/LogAnalyzer.java
package loganalazyer;

import java.util.List;

public class LogAnalyzer {
    private LogSource logSource;
    private Logger logger;

    public LogAnalyzer(LogSource logSource, Logger logger) {
        this.logSource = logSource;
        this.logger = logger;
    }

    public void analyze() {
        List<String> lines = logSource.getLines();

        for (String line : lines) {
            if (line.contains("Error: ")) {
                logger.logError(line);
            } else if (line.contains("Info: ")) {
                logger.logInfo(line);
            }
        }

    }
}

File: csharp/TestingExercises/TestingExercises/loganalyzer/LogAnalyzer.cs
namespace TestingExercises.loganalyzer
{
    public class LogAnalyzer
    {
        private readonly ILogSource _logSource;
        private readonly ILogger _logger;

        public LogAnalyzer(ILogSource logSource, ILogger logger)
        {
            _logSource = logSource;
            _logger = logger;
        }

        public void Analyze()
        {
            foreach(var line in _logSource.GetLines())
            {
                if (line.Contains("Error: "))
                {
                    _logger.LogError(line);
                }
                else if (line.Contains("Info: "))
                {
                    _logger.LogInfo(line);
                }
            }

        }
    }
}